kolibri_explore_plugin/collectionviews.py

```diff
@@ -803,6 +803,23 @@ def _get_latest_channels_for_all_collections(language_id):
     return set(channels.items())
 
 
+def _get_saved_collections_state(request):
+    collections_state_keys = (
+        "collection_name",
+        "collection_sequence",
+        "stage",
+    )
+    saved_state = request.session.get("COLLECTIONS_STATE")
+
+    if not saved_state:
+        return None
+
+    if not all(key in saved_state for key in collections_state_keys):
+        return None
+
```

Comment:
In practice, `CollectionDownloadManager.from_state` and `CollectionDownloadManager.to_state` are what define the format of this dictionary, so shouldn't this validation live in that class?

Reply:
:+1: Indeed it should! Fixed in the newest commit.